pkg/kv/kvpb/api.proto

```diff
@@ -1996,6 +1996,45 @@ message AddSSTableRequest {
   // also find and return the key at which the span after the added file span
   // is likely non-empty. See AddSSTableResponse.FollowingLikelyNonEmptySpanStart.
   bool return_following_likely_non_empty_span_start = 9;
+
+  // RemoteFile indicates that the span indicated by the request start and end 
+  // key of the file specified should be added. Data cannot be set if RemoteFile
+  // is non-empty, and many other request parameters such as collision/shadow
+  // checking, write-at-request-timestamp, or ingest-as-writes are unsupported.
+  //
+  // TODO(dt, msbutler, bilal): This is unsupported.
+  // TOOD(dt, msbutler, bilal): support sst_timestamp_to_request_timestamp.
+  message  RemoteFile {
+    string location = 1;
```

Comment:
I would support `s/location/locator`, as that matches the term we're using in Pebble to find a bucket. Maybe we can also add a requirement that `location` be a path that can be parsed to create an `ExternalStorage`?